hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/block/TestDeletedBlockLog.java

```diff
@@ -527,6 +533,22 @@ public void testDeletedBlockTransactions()
     Assertions.assertEquals(2, blocks.size());
   }
 
+  @Test
+  public void testDeletedBlockTransactionsOfDeletedContainer()
+      throws IOException, TimeoutException {
+    int txNum = 10;
+    List<DeletedBlocksTransaction> blocks;
+
+    // Creates {TXNum} TX in the log.
+    Map<Long, List<Long>> deletedBlocks = generateData(txNum,
```

Comment:
Containers are deleted only if block count is zero. In my opinion you could also have an integration testcase with underreplicated containers & going ahead deleting these blocks. This particular issue can happen then.